// learnfactory/learnfactory.js
/*
	This is an example code to build a factory module.
*/

var mydigitalstructure = require('mydigitalstructure')
var _ = require('lodash')
var moment = require('moment');

module.exports = 
{
	VERSION: '0.0.1',

	init: function (param)
	{
		var namespace = 'util';
		if (_.has(param, 'namespace')) {namespace = param.namespace}

		mydigitalstructure.add(
		{
			name: namespace + '-log',
			code: function (param, data)
			{
				var eventData = mydigitalstructure.get(
				{
					scope: 'learn',
					context: 'lambda',
					name: 'event'
				});

				mydigitalstructure.cloud.invoke(
				{
					object: 'core_debug_log',
					fields:
					{
						data: JSON.stringify(eventData),
						notes: 'Learn Lambda Log'
					},
					callback: namespace + '-log-saved',
					callbackParam: param
				});
			}
		});

		mydigitalstructure.add(
		{
			name: namespace + '-log-saved',
			code: function (param, response)
			{
				mydigitalstructure._util.message('learn-log factory');
				mydigitalstructure._util.message(param);
				mydigitalstructure._util.message(response);

				//This can be anything you want to return
				param = mydigitalstructure._util.param.set(param, 'logID', response.id);

				//Need to handle if a param.onComplete or param.onCompleteWhenCan passed
				mydigitalstructure._util.onComplete(param);
			}
		});
	}
}
